### ISSUE TITLE
It seems like celeste uses only 1 cpu core

### ISSUE THREAD

**Issue by loveisfoss** [hwittenborn/celeste#36]
It seems like celeste uses only 1 cpu core when working.

My laptop (CPU: intel 1255u) gets to 85 Celsius when celeste checks if there are any changes in my synced folder. Is this normal?

![Screenshot from 2023-02-08 21-45-12](https://user-images.githubusercontent.com/51087561/217648621-e97ea104-a3e8-4085-bfcd-743d6f509dd2.png)

**Comment by hwittenborn** [comment]
Celeste isn't able to do multi-core syncing yet, no. It's definitely something that I'd like to look into though, I have no doubt it could help to speed up transfers by a great deal.

I don't think I've ever checked my computer temperature for much of anything, so I wouldn't know what the norm is. Is your hardware fairly low-end, or is it still getting to that temperature with something decent?

**Comment by loveisfoss** [comment]
Thanks for the quick response and your good work!

It is a laptop from 2022 with an i7 CPU, bought a couple of months ago. I used it also on my desktop and I hear also the fan on my desktop spinning up when using celeste.

The thing is that my laptop is getting up to 90 Celsius when celeste is up and running and the fan gets at full speed almost at celeste start (and I put it in a way to not obstruct the airflow). When I play some light to moderate games, it doesn't get so loudy at all. When I just used the rclone command it was not so dramatic.

**Comment by pflanzenandi** [comment]
My Laptop goes from 40 (normal office work) to 60 Celsius; Celeste is using 12,5% CPU when running a sync.

**Comment by hwittenborn** [comment]
> When I just used the rclone command it was not at all so dramatic.

That's definitely strange, all Celeste really does is call Rclone commands and then update the progress in the UI. Do any other GTK applications make your computer heat up like Celeste is doing?

> My Laptop goes from 40 (normal office work) to 60 Celsius; Celeste is using 12,5% CPU when running a sync.

That's what my laptop is like too, though it's fairly low end (literally a Pentium iirc) so I kind of just expect it. What are the specs for your laptop like?

**Comment by pflanzenandi** [comment]
> That's what my laptop is like too, though it's fairly low end (literally a Pentium iirc) so I kind of just expect it. What are the specs for your laptop like?

XPS 9500; i5-10300H CPU @ 2.50GHz × 8; 16GB Ram

**Comment by hwittenborn** [comment]
Yeah that's definitely not right, does this kind of behavior happen for any other GTK apps for you @pflanzenandi?

**Comment by pflanzenandi** [comment]
> Yeah that's definitely not right, does this kind of behavior happen for any other GTK apps for you @pflanzenandi?

No, I rarely see apps using > 5%  CPU.

**Comment by hwittenborn** [comment]
> I did notice that Celeste keeps checking the exclusion list pretty much constantly for my Google Drive connection though, which doesn't seem right.

It's a bit of a flaw with how that part of Celeste was designed - Celeste writes out the exclusion file immediately after you edit it in the UI, so in the event of a crash you don't lose any modifications you've made. There's no caching done on what those file's content were though, so something like that could definitely be implemented.

> Looking in system monitor the .tmp file associated with it is using around 6.5% of my CPU.

What temp file is that? If you read it does it look like a binary file, and does killing that process kill the tray and then the main application?

I wouldn't think the file reading would cause such a CPU spike though, I'll do some checking to see what might be up though.

**Comment by vaelfonia** [comment]
Its name is `.tmp6bNR3S` - killing the process appears to do nothing as it instantly restarts, but it was certainly the source of the increased CPU usage.

**Comment by pflanzenandi** [comment]
When killing the .tmp process the tray icon closes, but Celeste remains open.

![Screenshot-20230209212127-985x664](https://user-images.githubusercontent.com/60046516/217928509-e202d0d1-53c9-4e4e-8207-d8be2f7f016b.png)

**Comment by vaelfonia** [comment]
> When killing the .tmp process the tray icon closes, but Celeste remains open.
> 
> ![Screenshot-20230209212127-985x664](https://user-images.githubusercontent.com/60046516/217928509-e202d0d1-53c9-4e4e-8207-d8be2f7f016b.png)

Yeah, essentially the same here except I lack an icon despite having TrayIcons Reloaded running.
